bin/apibuilder

```diff
@@ -108,17 +108,20 @@ end
 def head_of_latest_uploaded_version(org, application, client)
   found = nil
   0.upto(99).each do |i|
-    version = ApibuilderCli::Git.safe_describe(i)
-    begin
-      apibuilder_version = client.versions.get_by_application_key_and_version(org, application, version)
-      found = i
-      break
-    rescue Io::Apibuilder::Api::V0::HttpClient::ServerError => e
-      # Continue looking when the version is not found; fail for other exceptions.
-      if e.code != 404
-        raise e
+    # Attempt both naming with branch and the legacy version without branch.
+    [ApibuilderCli::Git.generate_version(i), ApibuilderCli::Git.generate_version(i, true)].each do |version|
```

Comment:
do we really now know in advance how to name the branch? would prefer we had a single way to generate the version (and then could avoid the boolean param)

Reply:
This is to maintain backwards-compatibility, i.e. to identify versions that were created before the branch name was added to the version. Of course, I could not support this and simply expect people to manually clean up those legacy versions.

Also note that for master, I'm not including the branch name (so as to result in version names like `1.0.0` instead of `1.0.0-master`), so there is some potential conflict between a legacy naming for a non-master branch and the new naming logic that assumes that "no branch name == master".

If you have a better idea here, I'm definitely open to suggestion.

Reply:
perhaps we should add a one time command (e.g. cleanup_old_non_semver_versions) - and then people can run that once as they want?

Reply:
That could work, but it would be a pretty brute-force approach, i.e.

1. Pull all versions with no branch name
2. Look in the master branch for them
3. If not found, look for them in all other branches
4. Delete if not found in any other branch

I guess brute-force wouldn't be _terrible_ since it would be only run once.

And there is additionally the risk that a branch was local and not pushed to remote (and thus might not be found). I guess I could add an interactive option to, instead of Y/N on delete, rename the version with a manually-entered suffix. That way someone could put it "somewhere else" and not risk it getting deleted because it's not found on master.

Reply:
I just uploaded some additional commits, so you can see the logic around deleting branch-specific versions. If you think a brute-force one-off like I outline above makes sense, I will remove this boolean and add that as an option to the `clean` command (i.e. `--legacy` or something like that).

Reply:
@mbryzek I took a simpler approach in which you can specify the branch to move a version to. This will avoid attempting to search the entire list of branches, while still allowing people to preserve versions on other branches. Please see the latest commit.